ouroboros-network-framework/test/Test/Ouroboros/Network/Server2.hs

```diff
@@ -2124,6 +2132,171 @@ prop_connection_manager_valid_transition_order serverAcc (ArbDataFlow dataFlow)
                        (Script (toBearerInfo absBi :| [noAttenuation]))
                        maxAcceptedConnectionsLimit l
 
+-- | Property wrapping `multinodeExperiment`.
+--
+-- Note: this test validates connection manager counters using an upper bound
+-- approach since there's no reliable way to reconstruct the value that the
+-- counters should have at a given point in time. It's not quite possible to
+-- have the god's view of the system consistent with the information that's
+-- traced because there can be timing issues where a connection is in
+-- TerminatingSt (hence counted as 0) but in the God's view the connection is
+-- still being deleted (hence counted as 1). This is all due to the not having
```

Comment:
And what about using something like: [stable](https://github.com/input-output-hk/ouroboros-network/blob/master/ouroboros-network/test/Data/Signal.hs#L196), note also that we know the `TimeWaitTimeout` value so if the connection was closed at time `t` we know that it will be forgotten at `t + timeWaitTimeout`.

Reply:
Ah, that could work, I'd have to try it! But I prefer to do it in a different PR.

Reply:
Do you already know if that was feasable or not?

Reply:
I'd have to give it a try, I wouldn't be able to retrieve all counters information from the `ObservableNetworkState` but I could probably at least test the number of incoming/outgoing connections. In either case it is preferable to keep this upper bound test too.